src/main/java/it/smartcommunitylab/aac/webauthn/controller/WebAuthnRegistrationController.java

```diff
@@ -0,0 +1,195 @@
+
+            final WebAuthnIdentityService webAuthnIdentityService = webAuthnIdentityAuthority
+                    .getIdentityService(providerId);
+            final WebAuthnSubjectResolver subjectResolver = webAuthnIdentityService.getSubjectResolver();
+            Subject subject = subjectResolver
+                    .resolveByUserId(username);
+            WebAuthnIdentityProviderConfig providerCfg = registrationRepository.findByProviderId(providerId);
+            String realm = providerCfg.getRealm();
+            final WebAuthnRegistrationResponse response = rps.startRegistration(
+                    username,
+                    realm,
+                    displayName,
+                    subject);
+            return response;
+        } catch (Exception e) {
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Validates the attestation generated using the Credential Creation
+     * Options obtained through the {@link #generateAttestationOptions} controller
+     */
+    @Hidden
+    @PostMapping(value = "/auth/webauthn/attestations/{providerId}", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public String verifyAttestation(@RequestBody @Valid WebAuthnAttestationResponse body,
+            @PathVariable("providerId") String providerId) {
+        final ResponseStatusException invalidAttestationException = new ResponseStatusException(HttpStatus.BAD_REQUEST,
+                "Invalid attestation");
+        try {
+            final boolean canRegister = registrationRepository.findByProviderId(providerId)
+                    .getConfigMap()
+                    .isEnableRegistration();
+            if (!canRegister) {
+                throw new RegistrationException("registration is disabled");
+            }
+            WebAuthnRpService rps = webAuthnRpServiceReigistrationRepository.get(providerId);
+            WebAuthnIdentityProviderConfig providerCfg = registrationRepository.findByProviderId(providerId);
+            String realm = providerCfg.getRealm();
+
+            String key = body.getKey();
+            String attestationString = mapper.writeValueAsString(body.getAttestation());
+            PublicKeyCredential<AuthenticatorAttestationResponse, ClientRegistrationExtensionOutputs> pkc = PublicKeyCredential
+                    .parseRegistrationResponseJson(attestationString);
+
+            final String authenticatedUser = rps.finishRegistration(pkc, realm, key);
+            if (!StringUtils.hasText(authenticatedUser)) {
+                throw invalidAttestationException;
+            }
+            return "Welcome " + authenticatedUser + ". Next step is to authenticate your session";
+        } catch (Exception e) {
+            throw invalidAttestationException;
+        }
+    }
+
+    /**
+     * Checks if the provided object can be used as a valid username
+     */
+    private boolean isValidUsername(Object username) {
```

Comment:
same as authController.
- get a String in input
- define pattern once
- use SLUG_PATTERN
- use `matcher.matches()` in place of `find()`